homeassistant/components/person/__init__.py

```diff
@@ -75,217 +72,169 @@
-            seen_users.add(user_id)
+        return {**data, **update_data}
 
-            entities.append(Person(person_conf, True))
+    async def _validate_user_id(self, user_id):
+        """Validate the used user_id."""
+        if await self.hass.auth.async_get_user(user_id) is None:
+            raise ValueError("User does not exist")
 
-        if entities:
-            await self.component.async_add_entities(entities)
+        for persons in (self.data.values(), self.yaml_collection.async_items()):
+            if any(person for person in persons if person.get(CONF_USER_ID) == user_id):
+                raise ValueError("User already taken")
 
-        self.hass.bus.async_listen(EVENT_USER_REMOVED, self._user_removed)
+    async def _collection_changed(
+        self, change_type: str, item_id: str, config: Optional[dict]
+    ) -> None:
+        """Handle a collection change."""
+        if change_type != collection.CHANGE_REMOVED:
+            return
 
-    async def async_create_person(self, *, name, device_trackers=None, user_id=None):
-        """Create a new person."""
-        if not name:
-            raise ValueError("Name is required")
+        ent_reg = await entity_registry.async_get_registry(self.hass)
+        ent_reg.async_remove(ent_reg.async_get_entity_id(DOMAIN, DOMAIN, item_id))
 
-        if user_id is not None:
-            await self._validate_user_id(user_id)
 
-        person = {
-            CONF_ID: uuid.uuid4().hex,
-            CONF_NAME: name,
-            CONF_USER_ID: user_id,
-            CONF_DEVICE_TRACKERS: device_trackers or [],
-        }
-        self.storage_data[person[CONF_ID]] = person
-        self._async_schedule_save()
-        await self.component.async_add_entities([Person(person, True)])
-        return person
-
-    async def async_update_person(
-        self, person_id, *, name=_UNDEF, device_trackers=_UNDEF, user_id=_UNDEF
-    ):
-        """Update person."""
-        current = self.storage_data.get(person_id)
-
-        if current is None:
-            raise ValueError("Invalid person specified.")
-
-        changes = {
-            key: value
-            for key, value in (
-                (CONF_NAME, name),
-                (CONF_DEVICE_TRACKERS, device_trackers),
-                (CONF_USER_ID, user_id),
-            )
-            if value is not _UNDEF and current[key] != value
-        }
+async def filter_yaml_data(hass: HomeAssistantType, persons: List[dict]) -> List[dict]:
+    """Validate YAML data that we can't validate via schema."""
+    filtered = []
 
-        if CONF_USER_ID in changes and user_id is not None:
-            await self._validate_user_id(user_id)
+    for person_conf in persons:
+        user_id = person_conf.get(CONF_USER_ID)
 
-        self.storage_data[person_id].update(changes)
-        self._async_schedule_save()
+        if user_id is not None:
+            if await hass.auth.async_get_user(user_id) is None:
+                _LOGGER.error(
```

Comment:
Do we need to create a persistent notification for this?

Reply:
It really is a config validation error. Maybe `person` should be migrated to use `async_validate_config`, but not sure if that should be done in this PR.

Reply:
<img width="483" alt="image" src="https://user-images.githubusercontent.com/1444314/71734973-78119880-2e4d-11ea-81ce-4d3e6b93e51d.png">